content_scripts/vimium_frontend.coffee

```diff
@@ -162,6 +164,46 @@ initializeWhenEnabled = (newPassKeys) ->
     enterInsertModeIfElementIsFocused()
     installedListeners = true
 
+# If the user hasn't acted yet, and has specified that they don't want Vimium to lose focus, we should detect
+# whether the page has focused an element and automatically blur it again.
+FocusPrevention =
+  lastFocused: null
+  isStillFocused: true
+  allowFocusAfterEvents: ["keydown", "keypress", "keyup", "mousedown", "click", "mouseup"]
+
+  shouldUnfocus: ->
+    isEnabledForUrl and settings.isLoaded and settings.get "disableAutoFocus"
+  # Returns true if it has unfocused the last focused element, false otherwise.
+  tryUnfocus: ->
+    return false unless @lastFocused? and @shouldUnfocus()
+    @lastFocused.blur()
+    exitInsertMode()
+    true
+  unfocus: (event) ->
+    return true unless isFocusable event.target
+    FocusPrevention.lastFocused = event.target
+    DomUtils.suppressPropagation event if FocusPrevention.tryUnfocus()
+
+  # Either add or remove the event listeners responsible for stopping autofocus, depending on add.
+  modifyListeners: (add) ->
+    changeEventListener = if add then "addEventListener" else "removeEventListener"
+    document[changeEventListener] "focus", @unfocus, true
```

Comment:
Should have a `blur` listener that does `DomUtils.suppressPropagation event` so the page doesn't see an asymmetry between `blur`s and `focus`es.

Also, the `focus` and `blur` listeners should set `@isStillFocused` to `true` and `false` respectively.